spring-cloud-commons/src/test/java/org/springframework/cloud/util/InetUtilsTests.java

```diff
@@ -31,23 +34,43 @@
 
 	@Test
 	public void testGetFirstNonLoopbackHostInfo() {
+		assertNotNull(new InetUtils(new InetUtilsProperties()).findFirstNonLoopbackHostInfo());
 		assertNotNull(InetUtils.getFirstNonLoopbackHostInfo());
 	}
 
 	@Test
 	public void testGetFirstNonLoopbackAddress() {
+		assertNotNull(new InetUtils(new InetUtilsProperties()).findFirstNonLoopbackAddress());
 		assertNotNull(InetUtils.getFirstNonLoopbackAddress());
 	}
 
 	@Test
 	public void testConvert() throws Exception {
+		assertNotNull(new InetUtils(new InetUtilsProperties()).convertAddress(InetAddress.getByName("localhost")));
 		assertNotNull(InetUtils.convert(InetAddress.getByName("localhost")));
 	}
 
 	@Test
 	public void testHostInfo() throws Exception {
-		HostInfo info = InetUtils.getFirstNonLoopbackHostInfo();
+		HostInfo info = new InetUtils(new InetUtilsProperties()).findFirstNonLoopbackHostInfo();
 		assertNotNull(info.getIpAddressAsInt());
+
+		info = InetUtils.getFirstNonLoopbackHostInfo();
+		assertNotNull(info.getIpAddressAsInt());
+	}
+
+	@Test
+	public void testIgnoreInterface() {
```

Comment:
supernit: add javadoc or comment or rename test method to say this is to support docker overlay networks. Not sure the impact of blacklist vs whitelist on usability.